api/src/main/java/edu/cornell/mannlib/vitro/webapp/search/controller/SearchFilter.java

```diff
@@ -0,0 +1,337 @@
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jena.rdf.model.RDFNode;
+
+public class SearchFilter {
+
+    private static final String FILTER = "Filter";
+    private static final String RANGE_FILTER = "RangeFilter";
+
+    private String id;
+    private String name = "";
+    private String from = "";
+    private String to = "";
+    private String fromYear = "";
+    private String toYear = "";
+    private boolean isPublic = false;
+
+    private String min = "0";
+    private String max = "2000";
+    private int moreLimit = 30;
+    private int order = 0;
+    private String field = "";
+    private String endField = "";
+    private String inputText = "";
+    private boolean localizationRequired = false;
+    private boolean multivalued = false;
+    private boolean selected = false;
+    private boolean input = false;
+    private Map<String, FilterValue> values = new LinkedHashMap<>();
+
+    private boolean inputRegex = false;
+
+    private boolean facetsRequired;
+
+    private String type = FILTER;
+    private String rangeText = "";
+    private String rangeInput = "";
+    private boolean hidden = false;
+
+    public String getRangeInput() {
+        return rangeInput;
+    }
+
+    public void setRangeInput(String range) {
+        this.rangeInput = range;
+    }
+
+    public String getRangeText() {
+        return rangeText;
+    }
+
+    public SearchFilter(String id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(RDFNode rdfNode) {
+        if (rdfNode != null) {
+            name = rdfNode.asLiteral().getLexicalForm().trim();
+        }
+    }
+
+    public void setOrder(RDFNode rdfNode) {
+        if (rdfNode != null) {
+            order = rdfNode.asLiteral().getInt();
+        }
+    }
+
+    public Integer getOrder() {
+        return order;
+    }
+
+    public String getField() {
+        return field;
+    }
+
+    public String getEndField() {
+        return endField;
+    }
+
+    public void setEndField(String endField) {
+        this.endField = endField;
+    }
+
+    public void addValue(FilterValue value) {
+        values.put(value.getId(), value);
+    }
+
+    public FilterValue getValue(String name) {
+        return values.get(name);
+    }
+
+    public Map<String, FilterValue> getValues() {
+        return values;
+    }
+
+    public void setField(String fieldName) {
+        field = fieldName;
+    }
+
+    public boolean contains(String valueId) {
+        if (values.containsKey(valueId)) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
This should be simplified as `return values.containsKey(valueId)`?

Reply:
Fixed